ic0/src/lib.rs

```diff
@@ -8,5 +8,594 @@
     clippy::missing_safety_doc
 )]
 
-mod ic0;
-pub use crate::ic0::*;
+use std::mem::MaybeUninit;
```

Comment:
Could you provide a module level doc to summarize what methods are expected in this module?
"safe Rust translation of system API" is not an accurate/complete statement:
- `call_new` and `call_on_cleanup` are still `unsafe` functions.
- Several methods have a `*_uninit` twin, we should better have an explanation.